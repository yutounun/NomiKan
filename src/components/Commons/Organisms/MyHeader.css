.header {
  position: fixed;
  top: 0;
  color: black;
  padding: 20px 0px;
  font-size: 1.4em;
  font-weight: 600;
  width: 100%;
  background-color: white;
  text-align: center;
}
