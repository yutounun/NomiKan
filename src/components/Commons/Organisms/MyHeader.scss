@import "static/variables";

.header {
  position: fixed;
  top: 0;
  color: black;
  padding: 20px 0px;
  font-family: $title-font;
  font-size: 1.4em;
  font-weight: 600;
  width: 100%;
  background-color: white;
  text-align: center;
}
