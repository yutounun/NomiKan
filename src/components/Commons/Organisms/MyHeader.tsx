import { Stack, Typography } from "@mui/material";
import React from "react";
import GoBack from "@mui/icons-material/ArrowBackIos";

class Props {
  value?: string = "Title";

  hideGoBack?: boolean = false;

  reversedColor?: boolean = false;
}
function MyHeader({ value, hideGoBack, reversedColor }: Props) {
  const handleGoBack = () => {
    window.history.back();
  };
  return (
    <>
      {
        reversedColor
          ? (
            <Stack
              justifyContent="center"
              alignItems="center"
              data-testid="header"
              mt={2.8}
              sx={{
                backgroundColor: "#EBF3FF",
                position: "fixed",
                top: 0,
              }}
            >
              <Typography variant="h1" sx={{ color: "#4E31AA", fontSize: { lg: "40px", xs: "25px" } }}>{value}</Typography>
            </Stack>
          )
          : (
            <Stack
              justifyContent="center"
              alignItems="center"
              data-testid="header"
              mt={2.8}
              sx={{
                backgroundColor: "white",
                position: "fixed",
                top: 0,
              }}
            >
              <Typography variant="h1" sx={{ color: "#4E31AA", fontSize: { lg: "40px", xs: "25px" } }}>{value}</Typography>
            </Stack>
          )
}

      {/* In case go back btn isn't hidden */}
      {!hideGoBack && (
        <Stack
          justifyContent="left"
          alignItems="center"
          mt={3}
          sx={{
            backgroundColor: "#EBF3FF",
            position: "fixed",
            top: 0,
            left: { xs: 20, lg: 50 },
          }}
        >
          <GoBack onClick={handleGoBack} />
        </Stack>
      )}
    </>
  );
}

export default MyHeader;
