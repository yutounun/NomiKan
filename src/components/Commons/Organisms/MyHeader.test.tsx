import { render, screen } from "@testing-library/react";
import React from "react";
import MyHeader from "./MyHeader";

describe("MyHeader", () => {
  it("renders correctly", () => {
    render(<MyHeader />);
    expect(screen.getByTestId("header")).toBeInTheDocument();
  });

  test("renders the component with the correct prop", () => {
    const textProp = "NomiKan";
    render(<MyHeader value={textProp} />);
    const textElement = screen.getByText(textProp);
    expect(textElement).toBeInTheDocument();
  });

  it("takes a snapshot", () => {
    const { asFragment } = render(<MyHeader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
