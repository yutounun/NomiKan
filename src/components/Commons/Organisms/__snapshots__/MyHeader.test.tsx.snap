// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyHeader takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-wzdyko-MuiStack-root"
    data-testid="header"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1doh40h-MuiTypography-root"
    />
  </div>
  <div
    class="MuiStack-root css-9922yn-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIosIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
